lib/pure/collections/lists.nim

```diff
@@ -669,13 +719,33 @@ proc add*[T: SomeLinkedList](a: var T, b: T) {.since: (1, 5, 1).} =
   var tmp = b.copy
   a.addMoved tmp
 
+proc remove*[T](L: var SinglyLinkedList[T], n: SinglyLinkedNode[T]) =
+  ## Removes a node `n` from `L`.
+  ## Efficiency: O(n), unless `n` is the head element.
+  runnableExamples:
+    var
+      a = initSinglyLinkedList[int]()
+      n = newSinglyLinkedNode[int](5)
+    a.add(n)
+    assert 5 in a
+    a.remove(n)
+    assert 5 notin a
+
+  if n == L.head:
```

Comment:
this gives a SIGSEGV
```nim
when true:
  import std/lists
  from sequtils import toSeq
  var a = [3].toSinglyLinkedList
  let n = newSinglyLinkedNode[int](5)
  a.remove n
```

instead:
* raise `KeyError` when `n` is not found
* specify it in doc comment
* add corresponding example with `doAssertRaises`